c/CMakeLists.txt

```diff
@@ -419,6 +419,8 @@ if (BUILD_STATIC_LIBS)
   add_library (qpid-proton-core-static STATIC ${qpid-proton-core-src})
   target_compile_definitions(qpid-proton-core-static PUBLIC PROTON_DECLARE_STATIC)
   target_link_libraries (qpid-proton-core-static ${SSL_LIB} ${SASL_LIB} ${PLATFORM_LIBS})
+  set_target_properties(qpid-proton-core-static PROPERTIES
+    POSITION_INDEPENDENT_CODE ON)
```

Comment:
these static libs will not later link into final binaries in Dispatch together with other dynlibs without this

Reply:
@astitcher so the `POSITION_INDEPENDENT_CODE ON` is so far the only thing that is not upstreamable?

Reply:
@astitcher ideally, I want a cmake object library which dispatch then sucks in; I am not actually interested in creating an artifact here, I just want to put the proton symbols into dispatch... 

I got inspired here by

> For this to happen, first create the static library with an option that makes the compiler generate “Position Independent Code” for every object that is archived as a part of the static library. This is required, because, a dynamic library, by its definition, is a “shared library” - which means it is relocatable. In order to have object files in the static library packed within the dynamic library, it is important these objects are also relocatable - i.e., made position independent.
> https://blog.ramdoot.in/how-can-i-link-a-static-library-to-a-dynamic-library-e1f25c8095ef

Reply:
Otherwise I cant build libqpid-dispatch.so

Reply:
-fPIC can be forced onto CMake from commandline when needed, so it is not necessary to put it into CMakeLists.txt